protocols/gossipsub/src/behaviour.rs

```diff
@@ -1855,7 +1856,9 @@ where
                 "Received message on a topic we are not subscribed to: {:?}",
                 message.topic
             );
-            return;
+            if !self.config.flood_publish() {
```

Comment:
If we get a message on a topic that we are not subscribed to, i think we want to just drop it. Its an error and it shouldn't occur if the code is written correctly. We shouldn't then forward this message on, i dont think.